### PR TITLE
[recovery] Recovery Improvements

### PR DESCRIPTION
**Pull request by devdl11**

### CODE REVIEW

File: ion/src/shared/internal_storage.cpp [hunk 1]
```diff
@@ -336,18 +336,18 @@ void InternalStorage::destroyRecordsWithExtension(const char * extension) {
...
+  m_magicFooter = Magic;
+  m_delegate = nullptr;
+  m_lastRecordRetrieved = nullptr;
+  m_lastRecordRetrievedPointer = nullptr;
+  
   assert(m_magicHeader == Magic);
   assert(m_magicFooter == Magic);
   // Set the size of the first record to 0
-  overrideSizeAtPosition(m_buffer, 0);
+  // overrideSizeAtPosition(m_buffer, 0);
```
Comment (Yaya-Cout) [inline ion/src/shared/internal_storage.cpp]: Remove comment ?

Comment (devdl11) [reply ion/src/shared/internal_storage.cpp]: good idea ^^

File: bootloader/interface/menus/upsilon_recovery.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+void Bootloader::UpsilonRecoveryMenu::postOpen() {
+  // We override the open method
+  for (;;) {
+    uint64_t scan = Ion::Keyboard::scan();
+    if (scan == Ion::Keyboard::State(Ion::Keyboard::Key::Back)) {
+      while (Ion::Keyboard::scan() == Ion::Keyboard::State(Ion::Keyboard::Key::Back));
+      forceExit();
+      return;
+    } else if (scan == Ion::Keyboard::State(Ion::Keyboard::Key::OnOff)) {
+      Ion::Power::standby();
```
Comment (Yaya-Cout) [inline bootloader/interface/menus/upsilon_recovery.cpp]:
```suggestion
      Ion::Power::suspend();
```

File: bootloader/interface/menus/upsilon_recovery.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+namespace Bootloader {
+  class UpsilonRecoveryMenu : public Menu {
+    public:
+      UpsilonRecoveryMenu();
+      void setup() override;
+      void postOpen() override;
+  };
+}
+
+#endif
```
Comment (Yaya-Cout) [inline bootloader/interface/menus/upsilon_recovery.h]:
```suggestion
#endif

```

File: ion/src/device/bootloader/platform_info.cpp [hunk 4]
```diff
@@ -72,7 +75,9 @@ class UserlandHeader {
     m_upsilonMagicHeader(UpsilonMagic),
     m_UpsilonVersion{UPSILON_VERSION},
     m_osType(OSType),
-    m_upsilonMagicFooter(UpsilonMagic) { }
+    m_upsilonMagicFooter(UpsilonMagic),
```
Comment (Yaya-Cout) [inline ion/src/device/bootloader/platform_info.cpp:78]: Add an Upsilon extra magic header

File: ion/src/device/shared/drivers/bldata.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      uint32_t storageAddress() const { return m_storageAddress; }
+      uint32_t storageSize() const { return m_storageSize; }
+    private:
+      uint32_t m_header;
+      uint32_t m_storageAddress;
+      uint32_t m_storageSize;
+      uint32_t m_footer;
+  };
+}
+}
```
Comment (Yaya-Cout) [inline ion/src/device/shared/drivers/bldata.h:24]:
```suggestion
}

```

Comment (Yaya-Cout) [inline ion/src/device/shared/drivers/bldata.h]: Can you add newline ?
```suggestion
}

```

File: bootloader/interface/menus/upsilon_recovery.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+void Bootloader::UpsilonRecoveryMenu::postOpen() {
+  // We override the open method
+  for (;;) {
+    uint64_t scan = Ion::Keyboard::scan();
+    if (scan == Ion::Keyboard::State(Ion::Keyboard::Key::Back)) {
+      while (Ion::Keyboard::scan() == Ion::Keyboard::State(Ion::Keyboard::Key::Back));
+      forceExit();
+      return;
+    } else if (scan == Ion::Keyboard::State(Ion::Keyboard::Key::OnOff)) {
+      Ion::Power::suspend();
```
Comment (Yaya-Cout) [inline bootloader/interface/menus/upsilon_recovery.cpp:37]: Can you redraw the screen after it ?

Comment (devdl11) [reply bootloader/interface/menus/upsilon_recovery.cpp:37]: Do you mean, relaunch the recovery mode ?

Comment (Yaya-Cout) [reply bootloader/interface/menus/upsilon_recovery.cpp:37]: When the screen it powered down, it lost the frame buffer, you need to draw the menu a new time, not necessary by relaunching it.

**Pull request merged**